### PR TITLE
Retry fetching teams if no teams are found

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
There have been some reports saying that sometimes when opening the app you reach the screen saying you don't belong to any team.

This PR attempts to handle that by retrying fetching your teams to ensure that the user actually does not belong to any team before presenting the screen and if needed it will present the screen saying that there was an error fetching the teams.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12758

### CODE REVIEW

**Comment by sudheerDev** [comment]
This works but just concerned if there are any other bugs which could be in the same category i.e possibly because of cleanup job issues.

Lets keep an eye out for other similar bugs so we can revisit this if needed.

File: app/utils/teams.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+// @flow
+
+import type {Team} from 'mattermost-redux/types/teams';
+
+export function selectFirstAvailableTeam(teams: Array<Team>, primaryTeam: ?string): ?Team {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/teams.js]: we haven't but thought we could start :P

**Comment by enahum** [comment]
@sudheerDev yeah I think this can be caused of a race condition somewhere, that is the only thing that I can think of but I could not find it

File: app/utils/teams.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sudheerDev) [reply app/utils/teams.js]: Sounds fine to me but is everyone fine with this? I still see this discussion of flow vs typescript every now and then.

[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/teams.js]: 0/5

[the remark this answers is not in the document]
Comment (hmhealey) [reply app/utils/teams.js]: Are we happy with how Flow is working in redux? I like the idea of using it, and I assume we get easy support for it from react-native, but I don't think this is something we should just add on a whim without making a proper effort to use it. Have we thought it through enough to just add it in a random PR like this?

**Pull request merged**